testdata/script/stack_submit_multiple_merged_history.txt

```diff
@@ -0,0 +1,235 @@
+      "sha": "8526d1a7195abb635f28bc93155b9155b76f3881"
+    }
+  },
+  {
+    "number": 3,
+    "state": "open",
+    "title": "Add feature 3",
+    "body": "",
+    "html_url": "$SHAMHUB_URL/alice/example/change/3",
+    "head": {
+      "ref": "feature3",
+      "sha": "63b2d337c8172c9f79aec9c760efc95e3c0c8472"
+    },
+    "base": {
+      "ref": "feature2",
+      "sha": "9806160f3b27dbff81e496260d26fc32f5ee5cf0"
+    }
+  }
+]
+
+-- golden/start-comments.txt --
+- change: 1
+  body: |
+    This change is part of the following stack:
+
+    - #1 ◀
+        - #2
+            - #3
+
+    <sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>
+    <!-- gs:navigation comment -->
+- change: 2
+  body: |
+    This change is part of the following stack:
+
+    - #1
+        - #2 ◀
+            - #3
+
+    <sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>
+    <!-- gs:navigation comment -->
+- change: 3
+  body: |
+    This change is part of the following stack:
+
+    - #1
+        - #2
+            - #3 ◀
+
+    <sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>
+    <!-- gs:navigation comment -->
+-- golden/pr-1-merged.json --
+[
+  {
+    "number": 1,
+    "state": "closed",
+    "merged": true,
+    "title": "Add feature 1",
+    "body": "",
+    "html_url": "$SHAMHUB_URL/alice/example/change/1",
+    "head": {
+      "ref": "feature1",
+      "sha": "3246bf23d6c5ee1069a9678a1cf02de608366cab"
+    },
+    "base": {
+      "ref": "main",
+      "sha": "59d3064874a8d0a003755556a982d1fe8adb59dc"
+    }
+  },
+  {
+    "number": 2,
+    "state": "closed",
+    "merged": true,
+    "title": "Add feature 2",
+    "body": "",
+    "html_url": "$SHAMHUB_URL/alice/example/change/2",
+    "head": {
+      "ref": "feature2",
+      "sha": "9806160f3b27dbff81e496260d26fc32f5ee5cf0"
+    },
+    "base": {
+      "ref": "feature1",
+      "sha": "3246bf23d6c5ee1069a9678a1cf02de608366cab"
+    }
+  },
+  {
+    "number": 3,
+    "state": "open",
+    "title": "Add feature 3",
+    "body": "",
+    "html_url": "$SHAMHUB_URL/alice/example/change/3",
+    "head": {
+      "ref": "feature3",
+      "sha": "1ee6dd04d1495e0815e85f9e3edc3912f6c81c81"
+    },
+    "base": {
+      "ref": "main",
+      "sha": "59d3064874a8d0a003755556a982d1fe8adb59dc"
+    }
+  }
+]
+
+-- golden/pr-1-merged-comments.txt --
+- change: 1
+  body: |
+    This change is part of the following stack:
+
+    - #1 ◀
+        - #2
+            - #3
+
+    <sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>
+    <!-- gs:navigation comment -->
+- change: 2
+  body: |
+    This change is part of the following stack:
+
+    - #1
+        - #2 ◀
+            - #3
+
+    <sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>
+    <!-- gs:navigation comment -->
+- change: 3
+  body: |
+    This change is part of the following stack:
+
+    - #2
+        - #3 ◀
```

Comment:
This one is wrong - the first PR is lost in the history if both 1 and 2 are detected as merged in the same sync.